app/controllers/application_controller.rb

```diff
@@ -16,6 +18,10 @@ def routing_error!
     raise ActionController::RoutingError, %|(No route matches [GET] "#{request.path}")|
   end
 
+  def validate_page_params(page_param, total_pages)
+    raise PageNotFound if page_param.to_i > total_pages
```

Comment:
Can you make response 404 for the following case?
* `page_param` is not integer
* `page_param` ia a negative value